lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -676,6 +676,106 @@ class DecomposeAtenTOp : public OpRewritePattern<AtenTOp> {
 };
 } // namespace
 
+// Decompose aten.repeat into aten.expand and aten.view ops.
+//
+// Ref: https://pytorch.org/docs/stable/generated/torch.Tensor.repeat.html
+//
+// For shape [S1, S2, S3] and repeats [M0, M1, M2, M3]
+//     MS0 = M0; MS1 = M1 * S1; MS2 = M2 * S2; MS3 = M3 * S3
+//
+// def aten_repeat(self, repeats):
+//     sizes = self.size()
+//     new_sizes0 = []
+//     new_sizes1 = []
+//     new_sizes2 = []
+//     leading_rank = repeats.size() - sizes.size()
+//     for r in range(leading_rank):
+//         new_sizes0.append(1)
+//         new_sizes1.append(repeats[r])
+//         new_sizes2.append(repeats[r])
+//
+//     for s, m in zip(sizes, repeats[leading_rank:]):
+//         n_sizes0 += [1, s]
+//         n_sizes1 += [m, s]
+//         n_sizes2 += [m * s]
+//     return self.view(new_sizes0).expand(new_sizes1).view(n_sizes2)
+//
+namespace {
+class DecomposeAtenRepeatOp : public OpRewritePattern<AtenRepeatOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenRepeatOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.self();
+    MLIRContext *context = op.getContext();
+    int64_t rank = getTensorRank(self);
+    if (rank < 0)
+      return rewriter.notifyMatchFailure(op, "Unimplemented: unranked tensor");
+
+    SmallVector<Value> repeats;
+    if (!getListConstructElements(op.repeats(), repeats))
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented: repeats not list of Scalar");
+
+    if (rank > repeats.size()) {
+      return rewriter.notifyMatchFailure(
+          op, "repeats are not matched with self's rank");
+    }
+
+    SmallVector<Value> newSizes0, newSizes1, newSizes2;
+    Value one = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    int64_t leadingRank = repeats.size() - rank;
+    for (int i = 0; i < leadingRank; ++i) {
```

Comment:
Since `leadingRank` >= 0 hence can use unsigned instead of int. The same can be used below.